modules/kubernetes/external-dns/main.tf

```diff
@@ -16,7 +16,15 @@ terraform {
 }
 
 locals {
-  values = templatefile("${path.module}/templates/values.yaml.tpl", { provider = var.dns_provider, sources = var.sources, azure_tenant_id = var.azure_tenant_id, azure_subscription_id = var.azure_subscription_id, azure_resource_group = var.azure_resource_group })
+  values = templatefile("${path.module}/templates/values.yaml.tpl", {
```

Comment:
What about loading different values-files based on the cloud provider?

Reply:
I am fine with doing it this way for now, we can come back to this later if the values file becomes to crowded.